tools/runfiles/Bazel/Runfiles.hs

```diff
@@ -0,0 +1,60 @@
+-- | This module enables finding data dependencies ("runfiles") of Haskell
+-- binaries at runtime.
+--
+-- For more information, see: https://github.com/bazelbuild/bazel/issues/4460
+--
+-- Note: this does not currently support the RUNFILES_MANIFEST environmental
+-- variable.  However, that's only necessary on Windows, which rules_haskell
+-- doesn't support yet.
+--
+-- Additionally, this is not yet supported by the REPL.
+module Bazel.Runfiles
+    ( Runfiles
+    , getRunfiles
+    , rlocation
+    , runfilesEnv
+    ) where
+
+import System.Directory (doesDirectoryExist)
+import System.Environment (getExecutablePath, lookupEnv)
+import System.FilePath (FilePath, (</>), (<.>))
+
+-- | A path to a directory tree containing runfiles for the given
+newtype Runfiles = Runfiles FilePath
+    deriving Show
+
+-- | Construct a path to a data dependency within the given runfiles.
+--
+-- For example: @rlocation \"myworkspace/mypackage/myfile.txt\"@
+rlocation :: Runfiles -> FilePath -> FilePath
+rlocation (Runfiles f) g = f </> g
+
+-- | Set environmental variables for locating the given runfiles directory.
+--
+-- Note that Bazel will set these automatically when it runs tests
+-- (@bazel test@).
+runfilesEnv :: Runfiles -> [(String, String)]
```

Comment:
What's the use case for this function? Should users set `RUNFILES_DIR` when it's not set? It was my understanding that `RUNFILES_DIR` is set iff the target is a data dependency of another. I don't know about tests.

Reply:
Added clarification in the comment; it's for non-test binaries to set manually, since from my testing it's only set automatically in "bazel test".